### PR TITLE
Fix failure when running BUILD_WITH_CONTAINER=1 make istioctl.completion

### LINKED ISSUE #20338: istio/istio: `BUILD_WITH_CONTAINER=1 make istioctl.completion` fails on MacOS
**Bug description**

Using MacOS,  `BUILD_WITH_CONTAINER=1 make istioctl.completion` fails
```
/work/out/darwin_amd64/istioctl collateral --bash && \
mv istioctl.bash /work/out/darwin_amd64/release/istioctl.bash
/bin/bash: /work/out/darwin_amd64/istioctl: cannot execute binary file: Exec format error
Makefile.core.mk:361: recipe for target '/work/out/darwin_amd64/release/istioctl.bash' failed
```

The issue is that the makefile running on Linux in the container is trying to run a MacOS binary crated during the build process.

**Expected behavior**

`BUILD_WITH_CONTAINER=1 make istioctl.completion` shown run successfully on a Mac

**Steps to reproduce the bug**

Prepare MacOS development as specified.

Run `BUILD_WITH_CONTAINER=1 make istioctl.completion`

**Version (include the output of `istioctl version --remote` and `kubectl version` and `helm version` if you used Helm)**

Current _master_ branch

**How was Istio installed?**

N/A

**Environment where bug was observed (cloud vendor, OS, etc)**

macOS Catalina

### PR DESCRIPTION
**Pull request by ericvn**
Fixes #20338

### CODE REVIEW

**Comment by ericvn** [comment]
`rm -rf out` (does a better `make clean` since the later doesn't remove the linux related directory)
`BUILD_WITH_CONTAINER=1 make istioctl.completion`
yields a successful run. A second `BUILD_WITH_CONTAINER=1 make istioctl.completion` without the clean is also successful.

File: Makefile.core.mk [hunk 1]
```diff
@@ -86,6 +86,15 @@ export ISTIO_BIN=$(GOBIN)
 
 export ISTIO_OUT:=$(TARGET_OUT)
 export ISTIO_OUT_LINUX:=$(TARGET_OUT_LINUX)
+
+# LOCAL_OUT should include architecture where we are currently running versus the desired.
```
Comment (howardjohn) [inline Makefile.core.mk]: I am a bit confused when LOCAL_OUT should be used and when ISTIO_OUT should be?

Comment (howardjohn) [reply Makefile.core.mk]: Oh is local-out is for running things, we never explicitly output things there?

Comment (ericvn) [reply Makefile.core.mk]: Correct. LOCAL_OUT is where you want run built artifacts from (ie, those artifacts should be of the OS/ARCH of the container if being run in the container or the local environment if not running in the container). I would have preferred to use *HOST* for specifying where we are running, *LOCAL* for the container, but we already were using *LOCAL* (copying from *TARGET*) so hard to come up with a good name.

**Pull request merged**